TorchDynamo: skip convolution fusion when convolution's padding is string

XiaobingSuper opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #88864
* __->__ #88794
* #88414
* #88412
* #88403


Currently,  the fusion convolution doesn't support the case when padding is a string, we will support it at the next step.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

XiaobingSuper committed:
TorchDynamo: skip convolution fusion when convolution's padding is string

[ghstack-poisoned]

XiaobingSuper committed:
Update on "TorchDynamo: skip convolution fusion when convolution's padding is string"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "TorchDynamo: skip convolution fusion when convolution's padding is string"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "TorchDynamo: skip convolution fusion when convolution's padding is string"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jgong5 commented on torch/_inductor/overrides.py:
```diff
@@ -486,7 +486,10 @@ def fuse_unary(gm: torch.fx.GraphModule):
                 if not eval_mode:
                     continue
-
+                if type(computation_node) in [nn.Conv2d] and isinstance(
+                    computation_node.padding, str
+                ):
```
Please add a comment (maybe with TODO) on why we don't support padding in `str` now.

XiaobingSuper replied:
Yes, one comment is added.

XiaobingSuper committed:
Update on "TorchDynamo: skip convolution fusion when convolution's padding is string"



Currently,  the fusion convolution doesn't support the case when padding is a string, we will support it at the next step.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "TorchDynamo: skip convolution fusion when convolution's padding is string"



Currently,  the fusion convolution doesn't support the case when padding is a string, we will support it at the next step.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "TorchDynamo: skip convolution fusion when convolution's padding is string"



Currently,  the fusion convolution doesn't support the case when padding is a string, we will support it at the next step.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "TorchDynamo: skip convolution fusion when convolution's padding is string"



Currently,  the fusion convolution doesn't support the case when padding is a string, we will support it at the next step.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "TorchDynamo: skip convolution fusion when convolution's padding is string"



Currently,  the fusion convolution doesn't support the case when padding is a string, we will support it at the next step.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper commented:
@pytorchbot merge
